common/app/com/commercetools/sunrise/framework/viewmodels/content/wishlist/WishlistViewModelFactory.java

```diff
@@ -0,0 +1,78 @@
+package com.commercetools.sunrise.framework.viewmodels.content.wishlist;
+
+import com.commercetools.sunrise.framework.viewmodels.SimpleViewModelFactory;
+import io.sphere.sdk.shoppinglists.LineItem;
+import io.sphere.sdk.shoppinglists.ShoppingList;
+import play.Configuration;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+/**
+ * This view model factory creates {@link WishlistViewModel} instances for wishlists {@link ShoppingList}.
+ */
+public class WishlistViewModelFactory extends SimpleViewModelFactory<WishlistViewModel, ShoppingList> {
+    private WishlistItemViewModelFactory wishlistItemViewModelFactory;
+
+    private final int recentlyAddedItems;
+    private final Locale locale;
+
+    @Inject
+    protected WishlistViewModelFactory(final WishlistItemViewModelFactory wishlistItemViewModelFactory,
+                                       final Configuration configuration,
+                                       final Locale locale) {
+        this.wishlistItemViewModelFactory = wishlistItemViewModelFactory;
+        this.recentlyAddedItems = configuration.getInt("wishlist.recentlyAddedItems");
+        this.locale = locale;
+    }
+
+    @Override
+    public final WishlistViewModel create(final ShoppingList wishlist) {
+        return super.create(wishlist);
+    }
+
+    @Override
+    protected WishlistViewModel newViewModelInstance(final ShoppingList wishlist) {
+        return new WishlistViewModel();
+    }
+
+    @Override
+    protected final void initialize(final WishlistViewModel viewModel, final ShoppingList wishlist) {
+        if (wishlist != null) {
+            fillTitle(viewModel, wishlist);
+
+            final List<LineItem> lineItems = wishlist.getLineItems();
+
+            fillQuantity(viewModel, lineItems);
+            fillLineItems(viewModel, lineItems);
+        }
```

Comment:
And pass the whole wishlist to those methods, we should not make assumptions of what the user may need to fill those fields, so it's better to pass exactly the same everywhere.